services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersSpec.groovy

```diff
@@ -55,87 +57,96 @@ class MetersSpec extends BaseSpecification {
             //unable to use #getExpectedBurst. For Centects there's special burst due to KBPS
-            assert it.meterEntries.every { it.burstSize == (long) ((DISCO_PKT_BURST * DISCO_PKT_SIZE) / 1024) }
-            assert it.meterEntries.every { ["KBPS", "BURST", "STATS"].containsAll(it.flags) }
-            assert it.meterEntries.every { it.flags.size() == 3 }
+            assert meters.meterEntries.every { it.burstSize == (long) ((DISCO_PKT_BURST * DISCO_PKT_SIZE) / 1024) }
+            assert meters.meterEntries.every { it.meterId <= MAX_SYSTEM_RULE_METER_ID }
+            assert meters.meterEntries.every { ["KBPS", "BURST", "STATS"].containsAll(it.flags) }
+            assert meters.meterEntries.every { it.flags.size() == 3 }
 
         }
     }
 
-    def "Default meters should express bandwidth in packet/s rather than kbps on the non-Centec switches"() {
+    def "Default meters should express bandwidth in pktps on non-Centec switches"() {
         requireProfiles("hardware") //TODO: Research how this behaves on OpenVSwitch
 
-        given: "All Openflow 1.3 compatible switches"
+        given: "All Openflow 1.3 compatible non-Centec switches"
         def switches = getNonCentecSwitches()
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
 
         expect: "Only the default meters should be present on each switch"
         switches.each { sw ->
             def meters = northbound.getAllMeters(sw.dpId)
             assert meters.meterEntries.size() == 2
             assert meters.meterEntries.every { it.rate == DISCO_PKT_RATE }
-            assert meters.meterEntries.every {
-                it.burstSize == getExpectedBurst(sw.dpId, DISCO_PKT_RATE)
-            }
+            assert meters.meterEntries.every { it.burstSize == getExpectedBurst(sw.dpId, DISCO_PKT_RATE) }
             assert meters.meterEntries.every { it.meterId <= MAX_SYSTEM_RULE_METER_ID }
             assert meters.meterEntries.every { ["PKTPS", "BURST", "STATS"].containsAll(it.flags) }
             assert meters.meterEntries.every { it.flags.size() == 3 }
         }
     }
 
     @Unroll
-    def "Meters are created/deleted when creating/deleting a flow (ignore_bandwidth=#ignoreBandwidth)"() {
-        given: "A switch with OpenFlow 1.3 support"
-        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+    def "Meters are created/deleted when creating/deleting a flow with ignore_bandwidth=#ignoreBandwidth \
+on a #switchType switch"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        given: "A #switchType switch with OpenFlow 1.3 support"
+        def sw = switches.first()
 
         when: "Get default meters from the switch"
         def defaultMeters = northbound.getAllMeters(sw.dpId)
         assert defaultMeters
 
-        and: "Create a flow"
+        and: "Create a single-switch flow"
```

Comment:
I think this detail is important enough to be reflected in the test name

Reply:
Fixed.